Source/JavaScriptCore/inspector/remote/RemoteConnectionToTarget.cpp

```diff
@@ -44,22 +44,28 @@ RemoteConnectionToTarget::~RemoteConnectionToTarget() = default;
 
 bool RemoteConnectionToTarget::setup(bool isAutomaticInspection, bool automaticallyPause)
 {
-    Locker locker { m_targetMutex };
-    if (!m_target)
-        return false;
+    RefPtr<RemoteControllableTarget> target;
+    TargetID targetIdentifier;
 
-    auto targetIdentifier = this->targetIdentifier().value_or(0);
+    {
+        Locker locker { m_targetMutex };
+        target = m_target.get();
+        if (!target)
+            return false;
+        targetIdentifier = this->targetIdentifier().value_or(0);
+    }
 
-    if (!m_target || !m_target->remoteControlAllowed()) {
+    if (!target->remoteControlAllowed()) {
         RemoteInspector::singleton().setupFailed(targetIdentifier);
+        Locker locker { m_targetMutex };
         m_target = nullptr;
-    } else if (auto* target = dynamicDowncast<RemoteInspectionTarget>(m_target)) {
-        target->connect(*this, isAutomaticInspection, automaticallyPause);
+    } else if (auto* inspectionTarget = dynamicDowncast<RemoteInspectionTarget>(*target) {
```

Comment:
`} else if (auto* inspectionTarget = dynamicDowncast<RemoteInspectionTarget>(*target)) {`

There is a missing parenthesis.